### DISCUSSION TITLE
Create first admin from code

### DISCUSSION THREAD

**Discussion by dnonnenm** [pocketbase/pocketbase#2933] [Q&A] [answered]
Hello

Is it possible to create the first admin user using code in the case of embedded pocketbase ?
I tried using OnAfterBootstrap callback, but it does not work (maybe the db is not yet ready).
So the work around I found was to use OnBeforeServe.

The main idea is to create the user from environment variables, passed from secrets in a kubernetes cluster.

Thank you very much

**Comment by ganigeorgiev** [comment, marked as answer]
You can create the first admin user programmatically by:

1. Using the `./pocketbase admin create test@example.com 1234567890` command.

2. Using a [migration](https://pocketbase.io/docs/migrations/) (Go or JS):
    - with Go:
    ```go
    package migrations

    import (
        "github.com/pocketbase/dbx"
        "github.com/pocketbase/pocketbase/daos"
        m "github.com/pocketbase/pocketbase/migrations"
        "github.com/pocketbase/pocketbase/models"
    )

    func init() {
        m.Register(func(db dbx.Builder) error {
            dao := daos.New(db)

            admin := &models.Admin{}
            admin.Email = "test@example.com"
            admin.SetPassword("1234567890")

            return dao.SaveAdmin(admin)
        }, nil)
    }
    ```
    
    - with JS:
    ```js
    migrate((db) => {
        const dao = new Dao(db);

        const admin = new Admin();
        admin.email = "test@example.com"
        admin.setPassword("1234567890")

        dao.saveAdmin(admin)
    }, null)
    ```
    
    _(more documentation about migrations will be available soon)_

**Reply by dnonnenm** [reply]
thank you very much @ganigeorgiev 
your advice is perfect.
I still need to check if the admin is already created, but it is ok; I will check the migrations to understand the mechanism.
